File: src/runtime2/cs/pytra/built_in/math.cs
using System;

namespace Pytra.CsModule
{
    // Python の math モジュール相当（sample 実行で使う最小集合）。
    public static class math
    {
        public const double pi = Math.PI;
        public const double e = Math.E;
        public const double tau = Math.PI * 2.0;

        public static double sin(double x) { return Math.Sin(x); }
        public static double cos(double x) { return Math.Cos(x); }
        public static double tan(double x) { return Math.Tan(x); }
        public static double asin(double x) { return Math.Asin(x); }
        public static double acos(double x) { return Math.Acos(x); }
        public static double atan(double x) { return Math.Atan(x); }
        public static double atan2(double y, double x) { return Math.Atan2(y, x); }
        public static double sqrt(double x) { return Math.Sqrt(x); }
        public static double exp(double x) { return Math.Exp(x); }
        public static double log(double x) { return Math.Log(x); }
        public static double floor(double x) { return Math.Floor(x); }
        public static double ceil(double x) { return Math.Ceiling(x); }
        public static double fabs(double x) { return Math.Abs(x); }
        public static double pow(double x, double y) { return Math.Pow(x, y); }
    }
}

File: src/runtime2/ts/pytra/math.ts
// Python の math モジュール互換（最小実装）。

export const pi: number = Math.PI;
export const e: number = Math.E;

export function sin(v: number): number { return Math.sin(v); }
export function cos(v: number): number { return Math.cos(v); }
export function tan(v: number): number { return Math.tan(v); }
export function sqrt(v: number): number { return Math.sqrt(v); }
export function exp(v: number): number { return Math.exp(v); }
export function log(v: number): number { return Math.log(v); }
export function log10(v: number): number { return Math.log10(v); }
export function fabs(v: number): number { return Math.abs(v); }
export function floor(v: number): number { return Math.floor(v); }
export function ceil(v: number): number { return Math.ceil(v); }
export function pow(a: number, b: number): number { return Math.pow(a, b); }
